src/base/QXmppMessage.cpp

```diff
@@ -1547,7 +1547,7 @@ void QXmppMessage::serializeExtensions(QXmlStreamWriter *writer, QXmpp::SceMode
             writer->writeDefaultNamespace(ns_xhtml_im);
             writer->writeStartElement(QStringLiteral("body"));
             writer->writeDefaultNamespace(ns_xhtml);
-            writer->writeCharacters(QStringLiteral(""));
+            writer->writeCharacters(QLatin1String(""));
```

Comment:
Hmm, could this line be removed completely? I mean, what's the effect of `writeCharacters` with an empty string anyway?

Reply:
Unfortunately not because the underlying device() is used directly below; the tests fail without it:
```
FAIL!  : tst_QXmppMessage::testXhtml() Compared values are not the same
   Actual   (data)        : "<message type=\"normal\"><body>hi!</body><html xmlns=\"http://jabber.org/protocol/xhtml-im\"><body xmlns=\"http://www.w3.org/1999/xhtml\"<p style=\"font-weight:bold\">hi!</p>/></html></message>"
   Expected (processedXml): "<message type=\"normal\"><body>hi!</body><html xmlns=\"http://jabber.org/protocol/xhtml-im\"><body xmlns=\"http://www.w3.org/1999/xhtml\"><p style=\"font-weight:bold\">hi!</p></body></html></message>"
   Loc: [/home/lnj/dev/qxmpp/tests/./util.h(58)]
```

Reply:
Ah wait but what does work is writing a null string.

Reply:
Ah, I see, so it's here for the writer to finish writing the opening `body` tag. Welp, this is so fragile, but I guess there are no better options.